SU2_CFD/src/solvers/CIncNSSolver.cpp

```diff
@@ -744,13 +735,22 @@ void CIncNSSolver::SetTauWall_WF(CGeometry *geometry, CSolver **solver_container
        * In incompressible flows, we can assume that there is no velocity-related
        * temperature change Prandtl: T+ = Pr*y+ ---*/
       su2double T_Wall = nodes->GetTemperature(iPoint);
+      su2double Conductivity_Wall = nodes->GetThermalConductivity(iPoint);
+
+      /*--- if a wall temperature was given, we compute the local heat flux using k*dT/dn ---*/
 
-      /*--- Extrapolate the pressure from the interior & compute the
-      wall density using the equation of state ---*/
+      su2double q_w = 0.0;
 
+      if (config->GetMarker_All_KindBC(iMarker) == ISOTHERMAL) {
+        const su2double T_n = nodes->GetTemperature(Point_Normal);
+        q_w = Conductivity_Wall * (T_Wall - T_n) / WallDistMod;
+      }  
+      else if (config->GetMarker_All_KindBC(iMarker) == HEAT_FLUX) {
+        q_w = config->GetWall_HeatFlux(Marker_Tag);  
+      }
```

Comment:
```suggestion
      }
      else if (config->GetMarker_All_KindBC(iMarker) == HEAT_TRANSFER) {
        const su2double Transfer_Coefficient = config->GetWall_HeatTransfer_Coefficient(Marker_Tag) * config->GetTemperature_Ref()/config->GetHeat_Flux_Ref();
        const su2double Tinfinity = config->GetWall_HeatTransfer_Temperature(Marker_Tag)/config->GetTemperature_Ref();
        q_w = Transfer_Coefficient * (Tinfinity - T_Wall);
      }
```

Reply:
That should do it, but it might be desirable to stay consistent with e.g. CIncNSSolver::BC_Wall_Generic and using the same variable names `Twall` instead of `T_Wall`, `Wall_HeatFlux` instead of `q_w`. I am not saying one naming is better or worse, but being consistent makes it easier to compare. 

Also one might want to put the `Get*(MarkerTag)` between the marker and vertex loop, is it doesn't inside the vertex loop